From 9475b98fcbf3636864063d90549c24c43ca189f7 Mon Sep 17 00:00:00 2001
From: Vignesh Venkatasubramanian <vigneshv@google.com>
Date: Fri, 19 Aug 2016 09:09:11 -0700
Subject: [PATCH 10/47] libvpx: Cherry-pick 8b4c315 from upstream

Description from upstream:
vp9_alloc_context_buffers: clear cm->mi* on failure

this fixes a crash in vp9_dec_setup_mi() via
vp9_init_context_buffers() should decoding continue and the decoder
resyncs on a smaller frame

Bug: 30593752
Test: builds
Change-Id: Iafbf1c4114062bf796f51a6b03be71328f7bcc69
Merged-In: Iafbf1c4114062bf796f51a6b03be71328f7bcc69
(cherry picked from commit 737c8493693243838128788fe9c3abc51f17338e)
---
 libvpx/vp9/common/vp9_alloccommon.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/libvpx/vp9/common/vp9_alloccommon.c b/libvpx/vp9/common/vp9_alloccommon.c
index 7dd1005..9826126 100644
--- a/libvpx/vp9/common/vp9_alloccommon.c
+++ b/libvpx/vp9/common/vp9_alloccommon.c
@@ -171,6 +171,7 @@ int vp9_alloc_context_buffers(VP9_COMMON *cm, int width, int height) {
   return 0;
 
  fail:
+  vp9_set_mb_mi(cm, 0, 0);
   vp9_free_context_buffers(cm);
   return 1;
 }
-- 
2.17.1

